scapy/layers/dns.py

```diff
@@ -193,19 +216,14 @@ def h2i(self, pkt, x):
             return b"."
         return x
 
-    def i2m(self, pkt, x):
-        if any((orb(y) >= 0xc0) for y in x):
-            # The value has already been processed. Do not process it again
-            return x
-
-        if not x or x == b".":
-            return b"\x00"
+    def any2i(self, pkt, x):
+        if isinstance(x, six.text_type):
+            if x and x[-1] != u".":
+                x += u"."
+        return StrField.any2i(self, pkt, x)
```

Comment:
Usual idiom is to use super().

Also, IMO it would be easier to turn it around:
```
x = super(StrField, self).any2i(pkt, x)
if x[-1] != b".":
    x += b"."
return x
```

Reply:
No, I don’t want to modify anything when bytes are provided. When you provide bytes, you know what you’re doing

Reply:
Works for me :-) Though this means that the same code `DNSRR(rrname='foo')` will behave differently on Python2 (no dot is added) and Python3 (dot is added).

Reply:
Yes but the UTF8 bug won’t be triggered, so it doesn’t matter

Reply:
If we ultimately decide not to handle Unicode, all we need to do is remove this `any2i` function + changes in `Scapy/compat.py`.

The rest of the PR is still a valid cleanup